app/controllers/Api2.scala

```diff
@@ -80,30 +88,34 @@ class Api2 (override val stores: DataStores, conf: Configuration, val authAction
     }
   }
 
-  def addAsset(atomId: String) = APIHMACAuthAction { implicit req =>
+  def addAsset(atomId: String) = APIHMACAuthAction.async { implicit req =>
     implicit val readCommand: Reads[AddAssetCommand] =
       (JsPath \ "uri").read[String].map { videoUri =>
         AddAssetCommand(atomId, videoUri, stores, youTube, req.user)
       }
 
-    parse(req) { command: AddAssetCommand =>
-      val atom = command.process()
-      Ok(Json.toJson(atom))
+    withAddAssetPermission(req) {
+      parse(req) { command: AddAssetCommand =>
+        val atom = command.process()
+        Ok(Json.toJson(atom))
+      }
     }
   }
 
 
   private def atomUrl(id: String) = s"/atom/$id"
 
-  def setActiveAsset(atomId: String) = APIHMACAuthAction { implicit req =>
+  def setActiveAsset(atomId: String) = APIHMACAuthAction.async { implicit req =>
     implicit val readCommand: Reads[ActiveAssetCommand] =
       (JsPath \ "youtubeId").read[String].map { videoUri =>
         ActiveAssetCommand(atomId, videoUri, stores, youTube, req.user)
       }
 
-    parse(req) { command: ActiveAssetCommand =>
-      val atom = command.process()
-      Ok(Json.toJson(atom))
+    withAddAssetPermission(req) {
```

Comment:
It might be nicer to chain the actions so `APIHMACAuthAction andThen withPermission(addAsset)`
It's a practise I've seen before when I was on Membership and it makes it easier to distinguish between the rules and the intent on the method. Happy to pair on this if you like?